src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts

```diff
@@ -157,9 +159,10 @@ export class LocalProcessExtensionHost implements IExtensionHost {
 		if (!this._messageProtocol) {
 			this._messageProtocol = Promise.all([
 				this._tryListenOnPipe(),
-				this._tryFindDebugPort()
-			]).then(([pipeName, portNumber]) => {
-				const env = objects.mixin(objects.deepClone(process.env), {
+				this._tryFindDebugPort(),
+				this._shellEnvironmentService.withShellEnv
+			]).then(([pipeName, portNumber, processEnv]) => {
+				const env = objects.mixin(processEnv, {
```

Comment:
Are we introducing some delay here? What if shell env is too slow? Or is it a fact that the process env is already resolved at this point?

Reply:
Yeah nothing changes with regards to resolving the shell env right on startup here:

https://github.com/microsoft/vscode/blob/c70ec95dd48ad70934b16ccbde14456129566c16/src/bootstrap-window.js#L34-L34

But I just realised that I should not await that when `VSCODE_SKIP_PROCESS_ENV_PATCHING` is set and pushed: https://github.com/microsoft/vscode/commit/c70ec95dd48ad70934b16ccbde14456129566c16